Find a functionally equivalent code:android.content.res.TypedArray.getColor:COMMENT
Method Modifier: public      
Comment:/**
 * Retrieve the color value for the attribute at <var>index</var>.  If
 * the attribute references a color resource holding a complex
 * {@link android.content.res.ColorStateList}, then the default color from
 * the set is returned.
 *
 * @param index Index of attribute to retrieve.
 * @param defValue Value to return if the attribute is not defined or
 * not a resource.
 *
 * @return Attribute color value, or defValue if not defined.
 */

@@ -1,4 +1,7 @@
 {
+    if (mRecycled) {
+        throw new RuntimeException("Cannot make calls to a recycled instance!");
+    }
     index *= AssetManager.STYLE_NUM_ENTRIES;
     final int[] data = mData;
     final int type = data[index + AssetManager.STYLE_TYPE];
@@ -13,18 +16,23 @@
             return csl.getDefaultColor();
         }
         return defValue;
+    } else if (type == TypedValue.TYPE_ATTRIBUTE) {
+        throw new RuntimeException("Failed to resolve attribute at index " + index);
     }
     throw new UnsupportedOperationException("Can't convert to color: type=0x" + Integer.toHexString(type));
 }

